src/Symfony/Component/Routing/Annotation/Route.php

```diff
@@ -69,7 +69,22 @@ public function __construct(
         } elseif (!\is_array($data)) {
             throw new \TypeError(sprintf('"%s": Argument $data is expected to be a string or array, got "%s".', __METHOD__, get_debug_type($data)));
         } elseif ([] !== $data) {
-            trigger_deprecation('symfony/routing', '5.3', 'Passing an array as first argument to "%s" is deprecated. Use named arguments instead.', __METHOD__);
+            $localizedPaths = [];
+            $deprecation = false;
+            foreach ($data as $key => $val) {
+                if (\in_array($key, ['path', 'name', 'requirements', 'options', 'defaults', 'host', 'methods', 'schemes', 'condition', 'priority', 'locale', 'format', 'utf8', 'stateless', 'env'])) {
+                    $deprecation = true;
+                } else {
+                    $localizedPaths[$key] = $val;
+                    unset($data[$key]);
+                }
+            }
```

Comment:
This basically means that I could mix a localized path with other route properties:

```php
#[Route(['en' => '/readme', 'de' => '/liesmich', 'methods' => ['GET']])]
```

This wasn't possible before and we should not start to allow it now.

Reply:
Yes I did that because of the deprecation that must be triggered. But maybe I miss something about this deprecation. Which use case should trigger the deprecation "Passing an array as first argument to "%s" is deprecated" please ?

Reply:
see my other comment